VM/vm_params2vm.py

```diff
@@ -0,0 +1,302 @@
+
+    msgs = []
+    logger.debug("Checking VM params file: {}".format(args.vm_params_path))
+
+    if not os.path.exists(args.vm_params_path):
+        raise FileNotFoundError(args.vm_params_path)
+
+    msgs.append(
+        (
+            args.name,
+            args.vm_params_path,
+            args.out_dir,
+            args.vm_threads,
+            qclogging.get_realisation_logger(logger, args.name).name,
+        )
+    )
+    return msgs
+
+
+def load_args(logger: Logger = qclogging.get_basic_logger()):
+
+    parser = ArgumentParser()
+    arg = parser.add_argument
+
+    arg("name", help="Name of the fault")
+
+    arg(
+        "vm_params_path",
+        help="path to vm_params.yaml",
+    )
+
+    arg("vm_root_dir", help="root directory to place VM files eg. Data/VMs")
+
+    arg("-n", "--nproc", help="number of processes", type=int, default=1)
+    arg(
+        "-t",
+        "--vm_threads",
+        "--threads",
+        help="number of threads for the VM generation",
+        type=int,
+        default=1,
+    )
+    arg("--vm-version", help="velocity model version to generate", default="1.65")
+    arg(
+        "--vm-topo",
+        help="topo_type parameter for velocity model generation",
+        default="BULLDOZED",
+    )
+
+    args = parser.parse_args()
+    args.vm_params_path = os.path.abspath(args.vm_params_path)
+    args.vm_root_dir = os.path.abspath(args.vm_root_dir)
+    return args
+
+
+if __name__ == "__main__":
+
+    logger = qclogging.get_logger("vm_params2vm")
+    qclogging.add_general_file_handler(
+        logger, os.path.join(os.getcwd(), "vm_params2vm_log.txt")
+    )
+    args = load_args(logger=logger)
+
+    # prepare to run
+    args.out_dir = os.path.join(args.vm_root_dir, args.name)
+    if not os.path.isdir(args.out_dir):
+        logger.debug(
+            "VM output directory {} does not exist. Creating it now.".format(
+                args.out_dir
+            )
+        )
+
+        os.makedirs(args.out_dir)
+
+    msg_list = load_msgs_vm_params(args, logger=logger)
+
+    # distribute work
+    logger.debug(
+        "Number of processes to use given as {}, number of tasks: {}.".format(
+            args.nproc, len(msg_list)
+        )
+    )
+    p = Pool(processes=args.nproc)
+    reports = p.starmap(main, msg_list)
+
+    # store summary
+    store_summary(
+        os.path.join(args.out_dir, "vm_params2vm_info.csv"), reports, logger=logger
+    )
+
+    # Hack to fix VM generation permission issue
+    hostname = platform.node()
+    if hostname.startswith(("maui", "mahuika", "wb", "ni")):  # Checks if is on the HPCF
+        logger.debug("HPC detected, changing folder ownership and permissions")
+        permission_cmd = ["chmod", "g+rwXs", "-R", args.out_dir]
+        subprocess.call(permission_cmd)
+        group_cmd = ["chgrp", constants.DEFAULT_ACCOUNT, "-R", args.out_dir]
+        subprocess.call(group_cmd)
```

Comment:
The account name should be stored somewhere, currently this doesn't work. 
Potentially it can be saved in the platform_config

Reply:
I think this permission issue can be circumvented if we place `umask 002` in the submission script.

Reply:
That would be great. Before removing this, it'd be good to test the umask command. 

However, that is probably beyond the scope of this PR.

Reply:
It is already in the shared bashrc.uceq for both Nurion and Maui. Yes, beyond the scope.

Reply:
Then the umask command doesn't work. So let's leave it in there rather than removing it.

Reply:
I disagree. Fixing permission, especially, if it is an HPC specific issue, should be handled by SLURM/PBS script.

Reply:
Agreed. In prior, we didn't have a slurm script for this as such so this was the best place for it.